### PR TITLE
Studio persistence v2 spec

### PR DESCRIPTION
**Pull request by ankitamritraj**
## How did you test this change?

- [x] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://github.com/finos/legend-studio/assets/19330019/f23b2773-d092-4ddb-b510-56b00f80b53d

### CODE REVIEW

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/transformation/pureProtocol/V1_DSL_Persistence_ProtocolHelper.ts [hunk 1]
```diff
@@ -890,6 +970,937 @@ export const V1_deserializeTargetShape = (
...
+
+const V1_serviceOutputTargetModelSchema = (
+  plugins: PureProtocolProcessorPlugin[],
+): ModelSchema<V1_ServiceOutputTarget> => {
+  return createModelSchema(V1_ServiceOutputTarget, {
+    serviceOutput: custom(
+      (val) => V1_serializeServiceOutput(val, plugins),
+      (val) => V1_deserializeServiceOutput(val, plugins),
+    ),
+    persistenceTarget: custom(
```
Comment (gayathrir11) [inline packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/transformation/pureProtocol/V1_DSL_Persistence_ProtocolHelper.ts:985]: Can you rearrange them in alphabetical order

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Persistence.ts [hunk 2]
```diff
@@ -35,7 +36,8 @@ export class Persistence
   documentation!: string;
   trigger!: Trigger;
   service!: PackageableElementReference<Service>;
-  persister!: Persister;
+  persister: Persister | undefined;
```
Comment (gayathrir11) [inline packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Persistence.ts:39]: Please modify this to
persister?: Persister | undefined;

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Persistence.ts [hunk 3]
```diff
@@ -45,7 +47,8 @@ export class Persistence
       this.documentation,
       this.trigger,
       this.service.valueForSerialization ?? '',
-      this.persister,
+      hashArray(this.serviceOutputTargets ?? []),
```
Comment (gayathrir11) [inline packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Persistence.ts:50]: Please make the similar modification as mentioned in protocol model

**Review by gayathrir11 — CHANGES_REQUESTED** [review]
Please add some unit tests here: packages\legend-extension-dsl-persistence\src\graph-manager\__tests__\DSL_Persistence_Roundtrip.test.ts

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Persistence.ts [hunk 4]
```diff
@@ -24,12 +24,14 @@ import {
...
 import { PERSISTENCE_HASH_STRUCTURE } from '../../../../../../../graph/DSL_Persistence_HashUtils.js';
 import type { V1_PersistenceTest } from './V1_DSL_Persistence_PersistenceTest.js';
+import { V1_ServiceOutputTarget as V1_ServiceOutputTarget } from './V1_DSL_Persistence_ServiceOutputTarget.js';
 
 export class V1_Persistence extends V1_PackageableElement implements Hashable {
   documentation!: string;
   trigger!: V1_Trigger;
   service!: string;
-  persister!: V1_Persister;
+  persister: V1_Persister | undefined;
```
[the remark this answers is not in the document]
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Persistence.ts]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Persistence.ts [hunk 5]
```diff
@@ -24,12 +24,14 @@ import {
...
+  serviceOutputTargets: V1_ServiceOutputTarget[] | undefined;
```
[the remark this answers is not in the document]
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Persistence.ts]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Persistence.ts [hunk 6]
```diff
@@ -39,7 +41,8 @@ export class V1_Persistence extends V1_PackageableElement implements Hashable {
       this.documentation,
       this.trigger,
       this.service,
-      this.persister,
+      hashArray(this.serviceOutputTargets ?? []),
```
[the remark this answers is not in the document]
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_Persistence.ts:44]: As discussed keeping it as empty array `[]`.

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Persistence.ts (hunk 2 above)
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Persistence.ts]: Updated.

File: packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Persistence.ts (hunk 3 above)
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph/metamodel/pure/model/packageableElements/persistence/DSL_Persistence_Persistence.ts:50]: As discussed keeping it as empty array [].

File: packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ActionIndicatorFields.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
```
[the remark this answers is not in the document]
Comment (ankitamritraj) [reply packages/legend-extension-dsl-persistence/src/graph-manager/protocol/pure/v1/model/packageableElements/persistence/V1_DSL_Persistence_ActionIndicatorFields.ts:2]: Added.

File: .changeset/lemon-zebras-talk.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+---
+---
```
Comment (gayathrir11) [inline .changeset/lemon-zebras-talk.md:2]: You don't need this. Can you remove this?

Comment (ankitamritraj) [reply .changeset/lemon-zebras-talk.md]: Removed.

File: packages/legend-manual-tests/src/__tests__/roundtrip-grammar/cases/DSL_PersistenceV2-unitemporal.pure [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        url: 'http://xyz.com';
+      },
+      Email
+      {
+        address: 'http://xyz.com';
+      }
+    ];
+  }
+  tests:
+  [
```
Comment (MauricioUyaguari) [inline packages/legend-manual-tests/src/__tests__/roundtrip-grammar/cases/DSL_PersistenceV2-unitemporal.pure:125]: could we have one of the tests have test grammar ?

**Review by MauricioUyaguari — DISMISSED** [review]

File: packages/legend-manual-tests/src/__tests__/roundtrip-grammar/cases/DSL_PersistenceV2-unitemporal.pure (hunk 9 above)
Comment (ankitamritraj) [reply packages/legend-manual-tests/src/__tests__/roundtrip-grammar/cases/DSL_PersistenceV2-unitemporal.pure:125]: Sure. I've added!

**Review by akphi — APPROVED** [review]

**Pull request merged**